middleware/checkfor.js

```diff
@@ -0,0 +1,48 @@
+checkFor([a,b,c,d,e])
```

Comment:
This is interesting, this is what @panamajake86 was potentially looking at doing. Might want to check in with him.

Reply:
I met with @panamajake86 today we came up with a clean middle ware function for this one.  will followup pr